### ISSUE TITLE
[Spike] - Verify past appointment fetch logic

### ISSUE THREAD

**Issue by JunTaoLuo** [department-of-veterans-affairs/va.gov-team#89552]
## Background/Request

It looks like the logic for retrieving the list of past appointment first fetch "cancelled" appointments [here](https://github.com/department-of-veterans-affairs/vets-website/blob/ffe4fcfe1c4527911843aef722d05cfaa79c96b9/src/applications/vaos/appointment-list/redux/actions.js#L374-L377) which is defined [here](https://github.com/department-of-veterans-affairs/vets-website/blob/47d6ebad7dd2f0793ce1c0a2d5602cf643c97e0f/src/applications/vaos/services/appointment/index.js#L84).

Later on, these fetched appointments are then filtered to remove all cancelled appointments via [`isValidPastAppointment`](https://github.com/department-of-veterans-affairs/vets-website/blob/ffe4fcfe1c4527911843aef722d05cfaa79c96b9/src/applications/vaos/services/appointment/index.js#L310-L325) in the [selectors](https://github.com/department-of-veterans-affairs/vets-website/blob/ffe4fcfe1c4527911843aef722d05cfaa79c96b9/src/applications/vaos/appointment-list/redux/selectors.js#L138-L165)

I think it's worth evaluating whether we can simplify this logic by specifying not to retrieve cancelled appointments in the original fetch.

### Goal

Determine if we can remove the filter for cancelled appointments in `isValidPastAppointment` by simply not fetching cancelled appointments in the first place when querying for past appointments.

### Requirements to Consider

---

## Tasks
- Answer the following questions:
  - Do we need to fetch cancelled appointments only to filter them out when retrieving past appointments?

## Time Box
_ hours

## Definition of Done
- Tasks are complete
- Research and recommendations are documented in this ticket

---
## How to configure this issue
- [ ] **Attached to an Epic** (what body of work is this a part of?)
- [ ] **Labeled with Practice Area** (`backend`, `frontend`, `devops`, `design`, `research`, `product`, `ia`, `qa`, `analytics`, `contact center`, `research`, `accessibility`, `content`)

**Comment by Bren22va** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/Appointments-Team-603fdef281af6500110a1691/issues/department-of-veterans-affairs/va.gov-team/89552?planning-poker) with Zenhub @cferris32 @jenniemc @JunTaoLuo @ryanshaw @simiadebowale @vbahinwillit

**Comment by Bren22va** [comment]
Please [add your planning poker estimate](https://app.zenhub.com/workspaces/Appointments-Team-603fdef281af6500110a1691/issues/department-of-veterans-affairs/va.gov-team/89552?planning-poker) with Zenhub @vbahinwillit

**Comment by vbahinwillit** [comment]
**Question**:
Do we need to fetch cancelled appointments only to filter them out when retrieving past appointments?

**Answer**:
No, we do not need to fetch 'canceled' past appointments when displaying the current past appointments page. There used to be a past appointment layout that displayed canceled appointments. This design decision boils down to:

- Select all appointment statuses at one time (current approach)
- Select appointments by status as needed.

I do not know what the metrics are for how users use the past appointments page but here are some use cases.

If the user navigates to the past appointments page from the upcoming appointments page and doesn't change the initial query (Past 3 months), advantage goes to select all appointment statuses at one time since all appointments statuses were returned when displaying the upcoming appointments page thus resulting in 1 api call.

If the user never goes to the past appointments page, advantage goes to select appointments by status since canceled appointments would not have been retrieved when displaying the upcoming appointments pages thus resulting in less data returned in the api call.

If the user visits the past appointments page and changes the initial query, advantage goes to select appointments by status since each date range change requires a new api call.

**Will making this change simplify the code?**
Yes. Refactoring code is always a good thing IMO.

**Comment by JunTaoLuo** [comment]
Thanks @vbahinwillit for looking into this. I want to clarify one bit which is that the initial query logic for upcoming appointments is a little different. It calls [`fetchFutureAppointments`](https://github.com/department-of-veterans-affairs/vets-website/blob/9bfd080c855c641a0613fec7b1f76a7f972971ac/src/applications/vaos/appointment-list/components/UpcomingAppointmentsList.jsx#L44C18-L44C41) which fetches appointments ('booked','arrived', 'fulfilled','cancelled') [for the past 30 days](https://github.com/department-of-veterans-affairs/vets-website/blob/9bfd080c855c641a0613fec7b1f76a7f972971ac/src/applications/vaos/appointment-list/redux/actions.js#L165-L172) and requests ('proposed','cancelled',) [for the past 120 days](https://github.com/department-of-veterans-affairs/vets-website/blob/9bfd080c855c641a0613fec7b1f76a7f972971ac/src/applications/vaos/appointment-list/redux/actions.js#L176C11-L184). Both of these are different from the logic used for past appointments which calls [`fetchPastAppointments`](https://github.com/department-of-veterans-affairs/vets-website/blob/9bfd080c855c641a0613fec7b1f76a7f972971ac/src/applications/vaos/appointment-list/components/PastAppointmentsList/index.jsx#L115C9-L115C30) that initially fetches appointments ('booked','arrived', 'fulfilled','cancelled') [for the past 3 months](https://github.com/department-of-veterans-affairs/vets-website/blob/9bfd080c855c641a0613fec7b1f76a7f972971ac/src/applications/vaos/appointment-list/components/PastAppointmentsList/index.jsx#L29-L38).

Basically I don't see any overlap in logic between upcoming and past appointment fetches at all since past 120 days != past 3 months anyways.

**Comment by vbahinwillit** [comment]
@JunTaoLuo - You are correct. I could have sworn that the initial request for appointments on the upcoming page was populating the past appointments collection as well but that actually doesn't make sense. So, removing the 'cancelled' appointments from the 'fetchPastAppointments' request should work just fine.

**Comment by ldelacosta** [comment]
@simiadebowale @JRRoof - I'm going to close this ticket. We could look to into this spike to see how we can improve fetching the Past appointment list.
